src/client/pages/AssessmentSection/Descriptions/CommentableDescription/Description/DataSources/DataSourceColumn/ColumnTypeOfDataSource.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import classNames from 'classnames'
+
+import { DataSource, dataSourceType } from '@meta/assessment'
+import { DataSourceDescription } from '@meta/assessment/description/nationalDataDataSourceDescription'
+
+import Autocomplete from '@client/components/Autocomplete'
+import DataColumn from '@client/components/DataGrid/DataColumn'
+import VerticallyGrowingTextField from '@client/components/VerticallyGrowingTextField'
+
+type Props = {
+  disabled: boolean
+  dataSourceValue: DataSource
+  onChange: (key: string, value: string) => void
+}
+
+const TextInput: React.FC<Props> = (props: Props) => {
```

Comment:
Subcomponents kept in same value, only used here and only used to tidy and separate logic from parent